### PR TITLE
fix(NotificationItemNew): tracking fix & accessibility improvement

### PR DESCRIPTION
**Pull request by saghdaey**
prevent both the action of opening link + tracking opening link, when the notification dismiss icon is pressed
make aria-label on notifications clearer

 - previously the whole Notification link content was being read via screen reader i.e. "Bell icon. {notification text} dismiss Icon" then when you tabbed again, it would say "dismiss icon" since that 'x' = another tabbable element.
 - fixes this bug: https://codecademy.atlassian.net/browse/EGG-912

### PR Checklist

- [ ] Related to designs:
- [x] Related to JIRA ticket: EGG-888 & EGG-912
- [x] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### CODE REVIEW

File: packages/gamut/src/NotificationListNew/NotificationItemNew.tsx [hunk 1]
```diff
@@ -52,12 +53,12 @@ export const NotificationItemNew: React.FC<NotificationItemNewProps> = ({
       return <StyledImg src={imageUrl} alt="" />;
     }
     if (type === 'marketing_blast') {
-      return <Megaphone height={48} width={48} aria-label="marketing update" />;
+      return <Megaphone aria-hidden height={48} width={48} />;
```
Comment (saghdaey) [inline packages/gamut/src/NotificationListNew/NotificationItemNew.tsx:56]: upon further consideration, hiding these icons from the screen reader. i don't think the learner needs to know the notification is from marketing or curriculum, these are more for our own internal tracking

File: packages/gamut/src/NotificationListNew/NotificationItemNew.tsx [hunk 2]
```diff
@@ -44,6 +44,7 @@ export const NotificationItemNew: React.FC<NotificationItemNewProps> = ({
 
   const dismissNotification = (event: React.MouseEvent) => {
     event.preventDefault();
+    event.stopPropagation();
```
Comment (saghdaey) [inline packages/gamut/src/NotificationListNew/NotificationItemNew.tsx:47]: need both of these to stop the click on the dismiss icon from
- opening the link 
- tracking the notification click
https://codecademy.atlassian.net/browse/EGG-888

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.2.16-alpha.99b90d.0<br/>@codecademy/gamut-labs@11.2.4-alpha.99b90d.0<br/>@codecademy/gamut@26.3.1-alpha.99b90d.0<br/>@codecademy/styleguide@30.6.1-alpha.99b90d.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.2.18-alpha.2fe634.0<br/>@codecademy/gamut-labs@11.4.1-alpha.2fe634.0<br/>@codecademy/gamut@26.4.1-alpha.2fe634.0<br/>@codecademy/styleguide@30.7.2-alpha.2fe634.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://605008fa802d5b0bbec9fd0f--gamut-preview.netlify.app>https://605008fa802d5b0bbec9fd0f--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/605008fa802d5b0bbec9fd0f>Deploy Logs</a></p>

File: packages/gamut/src/NotificationListNew/NotificationItemNew.tsx [hunk 3]
```diff
@@ -93,14 +94,17 @@ export const NotificationItemNew: React.FC<NotificationItemNewProps> = ({
...
           target="_blank"
           onClick={(event) => handleClick?.(event)}
         >
           {renderNotificationContent()}
         </StyledLink>
       ) : (
-        <div>{renderNotificationContent()}</div>
+        <div aria-label={`${text}, ${date} ago`}>
+          {renderNotificationContent()}
+        </div>
```
Comment (codecaaron) [inline packages/gamut/src/NotificationListNew/NotificationItemNew.tsx:107]: A label here might be redundant since the text should be visible :)

**Review by codecaaron — APPROVED** [review]

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.2.18-alpha.b484ed.0<br/>@codecademy/gamut-labs@11.4.1-alpha.b484ed.0<br/>@codecademy/gamut@26.4.1-alpha.b484ed.0<br/>@codecademy/styleguide@30.7.2-alpha.b484ed.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://605039f433eed3390076a2bb--gamut-preview.netlify.app>https://605039f433eed3390076a2bb--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/605039f433eed3390076a2bb>Deploy Logs</a></p>

**Review by JoshuaKGoldberg — APPROVED** [review]

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.2.19-alpha.26b33f.0<br/>@codecademy/gamut-labs@11.4.2-alpha.26b33f.0<br/>@codecademy/gamut@26.5.1-alpha.26b33f.0<br/>@codecademy/styleguide@30.8.1-alpha.26b33f.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6050b910688cf5c9153ffd60--gamut-preview.netlify.app>https://6050b910688cf5c9153ffd60--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6050b910688cf5c9153ffd60>Deploy Logs</a></p>

**Pull request merged**